hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java

```diff
@@ -265,7 +265,6 @@ private DiscoveryService initDiscoveryService(ClientConfig config) {
         if (isAutoDetectionEnabled && isEmptyDiscoveryStrategies(discoveryService)) {
             return null;
         }
-        discoveryService.start();
```

Comment:
As far as I understand, this start() in ClusterDiscoveryServiceBuilder is called once while constructing the client. 

The other start() is called when we try to connect to a cluster. So it can be called more than once because we can reconnect to a cluster, or switch cluster. So if we call start() in `CandidateClusterContext`, I think we can call start() more than once. 

When I check `discoveryService.start()` method, I think it is not a method that should be called more than once, or is it ? For example for DiscoverServices that has MulticastDiscoveryStrategy, it starts a thread each time start() is called. 

If it should be called once, we can call discoveryService.start() in ClusterDiscoveryServiceBuilder.java instead.

Reply:
It should be called each time you are about to connect a target cluster, because if you take a look at TcpClientConnectionManager#destroyCurrentClusterConnectionAndTryNextCluster, you will see that we destroy the discovery service, before trying the next one